### ISSUE TITLE
Migrating H2 to MySQL fails with "Column 'sizeX' cannot be null"

### ISSUE THREAD

**Issue by sunyal** [metabase/metabase#9756]
##- Ubuntu Bionic
- H2 and Cloud SQL MySQL 5.7
- Metabase version: 0.32.4
- Metabase hosting environment: Docker with a mounted database volume on Linux/Ubuntu Bionic (not running Docker during the migration)
- Metabase internal database: H2 (default)

- *Repeatable steps to reproduce the issue*
Unfortunately I cannot share the database that is producing the issue.
```
> java -jar metabase.jar load-from-h2 /path/to/metabase.db
log4j:ERROR Could not find value for key log4j.appender.metabase
log4j:ERROR Could not instantiate appender named "metabase".
04-10 18:17:07 INFO metabase.util :: Loading Metabase...
04-10 18:17:07 INFO metabase.util :: Maximum memory available to JVM: 3.7 GB
04-10 18:17:12 INFO util.encryption :: Saved credentials encryption is DISABLED for this Metabase instance. 🔓
 For more information, see https://metabase.com/docs/latest/operations-guide/start.html#encrypting-your-database-connection-details-at-rest
04-10 18:17:15 INFO metabase.driver :: Registered abstract driver :sql  🚚
04-10 18:17:19 INFO metabase.db :: Verifying mysql Database Connection ...
04-10 18:17:19 DEBUG plugins.classloader :: Setting current thread context classloader to NEWLY CREATED classloader clojure.lang.DynamicClassLoader@657d2501...
04-10 18:17:19 INFO metabase.driver :: Registered abstract driver :sql-jdbc (parents: :sql) 🚚
Load driver :sql-jdbc took 10 ms
04-10 18:17:19 INFO metabase.driver :: Registered driver :mysql (parents: :sql-jdbc) 🚚
Load driver :mysql took 79 ms
04-10 18:17:19 INFO metabase.driver :: Initializing driver :sql...
04-10 18:17:19 INFO metabase.driver :: Initializing driver :sql-jdbc...
04-10 18:17:19 INFO metabase.driver :: Initializing driver :mysql...
04-10 18:17:19 INFO metabase.db :: Verify Database Connection ...  ✅
04-10 18:17:19 INFO metabase.db :: Running Database Migrations...
04-10 18:17:19 INFO metabase.db :: Setting up Liquibase...
04-10 18:17:19 INFO metabase.db :: Liquibase is ready.
04-10 18:17:19 INFO metabase.db :: Checking if Database has unrun migrations...
04-10 18:17:21 INFO metabase.db :: Database Migrations Current ...  ✅
Temporarily disabling DB constraints...
[OK]
Transfering 9 instances of Database....[OK]
Transfering 29 instances of User....[OK]
Transfering 16 instances of Setting....[OK]
Transfering 8703 instances of Table.................................[OK]
Transfering 100839 instances of Field....................................................................................................................................................................................................................................................................................................................................................[OK]
Transfering 21870 instances of FieldValues............................................................................[OK]
Transfering 1 instances of Segment....[OK]
Transfering 392 instances of Revision.....[OK]
Transfering 1665 instances of ViewLog.........[OK]
Transfering 115 instances of Session....[OK]
Transfering 14 instances of Dashboard....[OK]
Transfering 52 instances of Card....[OK]
Transfering 1 instances of CardFavorite....[OK]
Transfering 47 instances of DashboardCard....BatchUpdateException:
 Message: (conn=40206) Column 'sizeX' cannot be null
 SQLState: 23000
 Error Code: 1048
java.sql.BatchUpdateException: (conn=40206) Column 'sizeX' cannot be null
	at org.mariadb.jdbc.MariaDbStatement.executeBatchExceptionEpilogue(MariaDbStatement.java:289)
	at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeBatch(MariaDbPreparedStatementClient.java:306)
	at clojure.java.jdbc$execute_batch.invokeStatic(jdbc.clj:591)
	at clojure.java.jdbc$execute_batch.invoke(jdbc.clj:584)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__12510.invoke(jdbc.clj:1039)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:842)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:782)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:1038)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:1024)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:1062)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:1042)
	at clojure.java.jdbc$insert_cols_BANG_.invokeStatic(jdbc.clj:1576)
	at clojure.java.jdbc$insert_cols_BANG_.invoke(jdbc.clj:1567)
	at clojure.java.jdbc$insert_multi_BANG_.invokeStatic(jdbc.clj:1633)
	at clojure.java.jdbc$insert_multi_BANG_.invoke(jdbc.clj:1601)
	at metabase.cmd.load_from_h2$insert_entity_BANG_$fn__47368.invoke(load_from_h2.clj:141)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invokeStatic(load_from_h2.clj:140)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invoke(load_from_h2.clj:123)
	at metabase.cmd.load_from_h2$load_data_BANG_.invokeStatic(load_from_h2.clj:154)
	at metabase.cmd.load_from_h2$load_data_BANG_.invoke(load_from_h2.clj:148)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_$fn__47488.invoke(load_from_h2.clj:235)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:799)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:834)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:782)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invokeStatic(load_from_h2.clj:228)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invoke(load_from_h2.clj:220)
	at clojure.lang.Var.invoke(Var.java:384)
	at metabase.cmd$load_from_h2.invokeStatic(cmd.clj:37)
	at metabase.cmd$load_from_h2.invoke(cmd.clj:30)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.cmd$run_cmd$fn__53180.invoke(cmd.clj:140)
	at metabase.cmd$run_cmd.invokeStatic(cmd.clj:140)
	at metabase.cmd$run_cmd.invoke(cmd.clj:136)
	at clojure.lang.Var.invoke(Var.java:388)
	at metabase.core$run_cmd.invokeStatic(core.clj:128)
	at metabase.core$run_cmd.invoke(core.clj:126)
	at metabase.core$_main.invokeStatic(core.clj:137)
	at metabase.core$_main.doInvoke(core.clj:133)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at metabase.core.main(Unknown Source)
Caused by: java.sql.SQLIntegrityConstraintViolationException: (conn=40206) Column 'sizeX' cannot be null
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:229)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:165)
	at org.mariadb.jdbc.MariaDbStatement.executeBatchExceptionEpilogue(MariaDbStatement.java:286)
	... 47 more
Caused by: java.sql.SQLException: Column 'sizeX' cannot be null
Query is: INSERT INTO report_dashboardcard ( `sizeX`, `updated_at`, `col`, `id`, `parameter_mappings`, `card_id`, `visualization_settings`, `dashboard_id`, `created_at`, `sizeY`, `row` ) VALUES ( <null>, '2018-11-06 20:02:25.43', 0, 12, '[{"parameter_id":"4c972b75","card_id":13,"target":["dimension",["template-tag","project"]]},{"parameter_id":"55fb9bb0","card_id":13,"target":["dimension",["template-tag","created_at"]]}]', 13, '{}', 1, '2018-04-23 22:49:07.548', <null>, 5 )
java thread: MariaDb-bulk-4
	at org.mariadb.jdbc.internal.util.LogQueryTool.exceptionWithQuery(LogQueryTool.java:126)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol$1.handleResultException(AbstractQueryProtocol.java:625)
	at org.mariadb.jdbc.internal.protocol.AsyncMultiRead.call(AsyncMultiRead.java:142)
	at org.mariadb.jdbc.internal.protocol.AsyncMultiRead.call(AsyncMultiRead.java:66)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
Command failed with exception: (conn=40206) Column 'sizeX' cannot be null
```

**Comment by camsaul** [comment]
@sunyal did you already upgrade to 0.32.4 before attempting to migrate that data?

**Comment by camsaul** [comment]
Fixed by #9758

**Comment by camsaul** [comment]
Hi @sunyal, this is fixed in Metabase 0.32.5 which has just been released. Please give it a try and let me know if it works for you!
